### PR TITLE
fix: enable write permissions for Gemini CLI write operations

### PR DESCRIPTION
**Pull request by theinterneti**
## 🎯 Root Cause Found!

**The write capability test failed because the workflow had insufficient permissions.**

---

## Problem

**Workflow Run #18978922410** failed with API error retry loop when attempting to create a file and PR.

**Root Cause**: Workflow permissions were set to `contents: 'read'` (lines 24 and 37), which prevents:
- File creation (`create_or_update_file` tool)
- Commits and pushes (`push_files` tool)
- Branch creation (`create_branch` tool)
- PR creation (`create_pull_request` tool)

**Error Pattern**:
```
[API Error: Exiting due to an error processing the @ command.]
Exiting due to an error processing the @ command.
```

Repeated every ~2 seconds for 15+ minutes (same as v0.18.0 timeout issue).

---

## Solution

Changed workflow permissions from `read` to `write`:

```yaml
# Before (❌ FAILED)
permissions:
  contents: 'read'  # Cannot create files, branches, or commits

# After (✅ SHOULD WORK)
permissions:
  contents: 'write'  # Required for file creation, commits, and branch creation
```

Also updated GitHub App token permissions:
```yaml
# Before (❌ FAILED)
permission-contents: 'read'

# After (✅ SHOULD WORK)
permission-contents: 'write'
```

---

## Changes

- ✅ Changed `contents: 'read'` to `contents: 'write'` in workflow permissions (line 24)
- ✅ Changed `permission-contents: 'read'` to `permission-contents: 'write'` in GitHub App (line 37)
- ✅ Added comments explaining why write permission is required

---

## Impact

This fix enables the following MCP server tools:

| Tool | Purpose | Status |
|------|---------|--------|
| `create_or_update_file` | Create/modify files | ✅ Now enabled |
| `push_files` | Commit and push changes | ✅ Now enabled |
| `create_branch` | Create new branches | ✅ Now enabled |
| `create_pull_request` | Create PRs | ✅ Now enabled |
| `delete_file` | Delete files | ✅ Now enabled |

---

## Security

**Write operations are still secure**:
- ✅ All changes go through PR review before merging
- ✅ All changes are auditable via Git history
- ✅ MCP server tools are explicitly whitelisted (lines 108-127)
- ✅ No workflow modification allowed
- ✅ No secrets access allowed
- ✅ All operations are logged in workflow runs

---

## Testing Plan

Once this PR is merged, retry the write capability test:

```markdown
@gemini-cli Create a test file at docs/test-gemini-write.md with the content 
"This file was created by Gemini CLI on October 31, 2025 to verify write capabilities. 
The GitHub MCP server v0.20.1 supports file creation, commits, and PR creation." 
Create a PR titled "test: verify Gemini CLI write capabilities".
```

**Expected Outcome**:
- ✅ New branch created
- ✅ File created with specified content
- ✅ Changes committed and pushed
- ✅ PR opened with title
- ✅ Comment posted with PR link
- ✅ Execution time: 1-2 minutes

---

## High-Value Use Cases Now Possible

With write permissions enabled, Gemini CLI can now:

1. **📝 Documentation Generation**
   ```markdown
   @gemini-cli Generate documentation for the CachePrimitive feature.
   Create a PR with updated README.md and new docs/primitives/cache.md file.
   ```

2. **🧪 Test File Creation**
   ```markdown
   @gemini-cli Generate comprehensive tests for the RetryPrimitive class.
   Create tests/test_retry_primitive.py with unit tests and edge cases.
   Create a PR.
   ```

3. **🔒 Dependency Updates**
   ```markdown
   @gemini-cli Update the requests library to version 2.31.0 to fix CVE-2023-XXXX.
   Update pyproject.toml and create a PR.
   ```

4. **🐛 Bug Fix PRs**
   ```markdown
   @gemini-cli Fix the null pointer exception in src/core/base.py line 42.
   Create a PR with the fix and add tests.
   ```

5. **♻️ Code Refactoring**
   ```markdown
   @gemini-cli Refactor src/core/base.py to use modern Python 3.11+ type hints.
   Update all type annotations and create a PR.
   ```

---

## Related

- **Issue #68**: Gemini CLI Timeout Issue (resolved with MCP v0.20.1)
- **PR #71**: MCP Server v0.20.1 Fix (merged)
- **Failed Workflow**: #18978922410 (write test with insufficient permissions)
- **Documentation**: `docs/gemini-cli-capabilities-analysis.md`
- **Documentation**: `docs/gemini-cli-integration-guide.md`

---

## Checklist

- [x] Root cause identified (insufficient permissions)
- [x] Fix implemented (contents: write)
- [x] Security implications reviewed (all changes go through PR review)
- [x] Testing plan documented (retry write capability test)
- [x] High-value use cases documented
- [ ] Write capability test passes (pending merge)
- [ ] Documentation updated with successful examples (pending test)

---

**This fix unlocks the full potential of Gemini CLI integration!** 🚀

---
Pull Request opened by [Augment Code](https://www.augmentcode.com/) with guidance from the PR author

### CODE REVIEW

File: .github/workflows/gemini-invoke.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     runs-on: 'ubuntu-latest'
     timeout-minutes: 15  # Prevent runaway executions (Karl Stoney's production: 3-5 min for PR reviews)
     permissions:
-      contents: 'read'
+      contents: 'write'  # Required for file creation, commits, and branch creation
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply .github/workflows/gemini-invoke.yml]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/theinterneti/TTA.dev/pull/73#discussion_r2482197697)
